### PR TITLE
docs: add release notes notifying change of default branch to `develop`

### PR DESCRIPTION
**Pull request by kwvg**
## Additional Information

Add a release note notifying the new default branch as `develop`.

## Checklist:

- [x] I have performed a self-review of my own code **(note: N/A)**
- [x] I have commented my code, particularly in hard-to-understand areas **(note: N/A)**
- [x] I have added or updated relevant unit/integration/functional/e2e tests **(note: N/A)**
- [x] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Review by PastaPastaPasta — APPROVED** [review]
utACK 8a66af25e802c96b493582630f155406485d6732

**Review by UdjinM6 — APPROVED** [review]
utACK 8a66af25e802c96b493582630f155406485d6732

File: doc/release-notes-6118.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Breaking Changes
+----------------
+
+* The Dash Core repository (`github.com/dashpay/dash`) will be using `develop` as its default branch. New clones
```
Comment (knst) [inline doc/release-notes-6118.md:4]: nit: `https://github.com/dashpay/dash` or `github.com:dashpay/dash` ?

**Review by knst — APPROVED** [review]
LGTM

File: doc/release-notes-6118.md (hunk 1 above)
Comment (kwvg) [reply doc/release-notes-6118.md:4]: Used the HTTPS URL as it's the most recognizable while omitting anything that doesn't strictly convey details about the path (the `https://`)

**Pull request merged**